### PR TITLE
Update the Makefile and the file hierarchy

### LINKED ISSUE #10: Update the file hierarchy and the Makefile to split the program and the library
To make the code easier to test, we should make a library `libirc.a`, and use everything that's in it in a restricted number of source files that will be the program.
For that, we should update the Makefile to have both target `libirc.a` and `ft_irc`.
Also, we should either create a sub-directory inside `src` to put the source files of the library or create a directory next to `src` for them.

### PR DESCRIPTION
**Pull request by JonathanDUFOUR**
This pull request is related to the issue #10.
The first files related to IRC commands have been added, and the main header file for the library has been added too. Files are now split into sub-directories, to easier separate the library files and the program files.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -6,66 +6,119 @@
...
 # **************************************************************************** #
 
 ######################################
 #              COMMANDS              #
 ######################################
-CXX			=	c++
-LINK		=	clang++
-MKDIR		=	mkdir -p
-RM			=	rm -rf
+   AR = ${shell which ar} rcs
```
Comment (Saphirac) [inline Makefile:16]: Why the change ?

File: Makefile [hunk 2]
```diff
@@ -6,66 +6,119 @@
...
-	CXXFLAGS	+=	-g
-	CXXFLAGS	+=	-DDEBUG=1
+	CXXFLAGS += -g -DDEBUG=1
 endif
 
 #######################################
 #                RULES                #
 #######################################
-${NAME}: ${OBJ}
+.PHONY:	all clean fclean re fre
```
Comment (Saphirac) [inline Makefile:113]: Are .PHONY not usually at the end?

**Review by Saphirac — COMMENTED** [review]
Please explain my comments before merging approval :)

File: Makefile (hunk 1 above)
Comment (JonathanDUFOUR) [reply Makefile:16]: This is a change that we made on the branch of @gle-mini some days ago, it allows us to have the absolute path to the commands in our variables, and not only their name. This is safer because it prevents aliases or functions or anything with the same name as the commands in question from hacking the correct behavior of our Makefile.

File: Makefile (hunk 2 above)
Comment (JonathanDUFOUR) [reply Makefile:113]: Even if we were used to putting it at the end of the file, it's a better practice to put it before the rules it contains. This is merely to optimize the `make` performance, winning only a few nanoseconds. To go deeper into the details, when `make` is invoked, it will read the `Makefile` until it finds the wanted target. Then, it will check whether the target is part of the `.PHONY` target. If it does, then `make` will execute the content of the target regardless of whether the file with the same name exists and is up to date. But knowing that `make` reads the `Makefile` only up to the wanted target, if it reaches it without reaching the `.PHONY`, it won't know that the wanted target is part of it. And `make` knows that it didn't read the entire file, so it will re-read it until either the end of the file is reached, or it finds a `.PHONY` target that contains the wanted target.
But now, suppose that the `.PHONY` that contains the wanted target is placed before it. This way, `make` will read the `.PHONY` before reaching the wanted target, and thus, it will already know that it is part of the `.PHONY` target. This will prevent `make` to re-read the `Makefile`.
Now that this is said, there exist 2 communities, those who prefer to have only 1 `.PHONY` target, so they will put it at the beginning only once, and those who prefer to have multiple `.PHONY` targets, so they will put each right before the target that they want to mark as phony. I prefer the first option.

**Review by Saphirac — APPROVED** [review]
I approve.
